assets/src/application/ui/layout/entityActionMenu/useEntityActions.ts

```diff
@@ -0,0 +1,77 @@
+import { pathOr, assocPath, omit, filter } from 'ramda';
+import uuidv4 from 'uuid/v4';
+
+import {
+	SubscribeFn,
+	Subscriptions,
+	Subscription,
+	SubscriptionsOptions,
+	EntityActions,
+	EntityActionsData,
+	UpdateSubscriptionProps,
+} from './types';
+
+const NAMESPACE = 'espresso';
+
+const useEntityActions = <Domain extends string, EntityType extends string, MenuKey extends string>(
+	domain: Domain
+): EntityActions<EntityType, MenuKey> => {
+	const subscribe: SubscribeFn<EntityType, MenuKey> = (callback, options): VoidFunction => {
+		// runtime check
+		if (typeof callback !== 'function') {
+			return;
+		}
+
+		const subscriptionId = uuidv4();
+
+		updateSubscription({ id: subscriptionId, callback, options, action: 'add' });
+
+		// to unsubscribe
+		return () => {
+			updateSubscription({ id: subscriptionId, action: 'remove' });
+		};
```

Comment:
Nice, the same pattern as with useEffect cleanup.

Reply:
Yes